src/corona_hakab_model/generation_analysis/population_analysis.py

```diff
@@ -48,15 +33,14 @@ def plot_circles_sizes(self):
             ax.set_ylabel("amount of circles")
             ax.set_xlabel("circle size")
             circles = self.population_data.social_circles_by_connection_type[con_type]
-            size_count = {}
+            size_count = defaultdict(int)
             for circle in circles:
                 size = circle.agent_count
                 if size not in size_count:
```

Comment:
since it is a defaultdict, you can get rid of this check (it will automatically get set to 0 for a new key)

Reply:
done